core/src/main/java/org/elasticsearch/search/suggest/SuggestionSearchContext.java

```diff
@@ -119,21 +112,13 @@ public void setShardSize(int shardSize) {
             }
             this.shardSize = shardSize;
         }
-        
-        public void setShard(int shardId) {
-            this.shardId = shardId;
-        }
 
-        public void setIndex(String index) {
-            this.index = index;
+        public void setShardContext(QueryShardContext context) {
```

Comment:
IMO, it would be cleaner to have `QueryShardContext` and `Suggester` as params in the ctor instead, since they are mandatory params? I have taken a stab at what I mean [here](https://gist.github.com/areek/afda767f37263558c4a6)